app/assets/stylesheets/layout/_top_menu.sass

```diff
@@ -166,103 +166,117 @@ $search-input-width-expanded: 300px
-  @include varprop(border-color, header-item-font-color)
-  border-radius: 4px
-  height: 31px
-  width: $search-input-width
-  outline: 0
-  font-size: 0.9rem
-  background: transparent
-  @include varprop(color, header-item-font-color)
-  -webkit-transition: width 0.2s ease-in-out
-  transition: width 0.2s ease-in-out
-  &::-ms-clear
-    margin-right: 5px
-    width: 20px
-  &:focus
-    background: white
-    border-width: 1px
-    border-style: solid
-    @include varprop(border-color, header-item-font-color)
+  .top-menu-search--button.-input-focused
     color: $header-search-field-font-color
-    &:hover
-      background: white
-  &:hover
-    background: unset
-    border-width: 1px
-    border-style: solid
-    @include varprop(border-color, header-item-font-color)
 
-input.top-menu-search--input.-expanded
-  width: $search-input-width-expanded
-  background: white
-  border-width: 1px
-  border-style: solid
-  @include varprop(border-color, header-item-font-color)
-  color: $header-search-field-font-color
-  &:hover
-    background: white
+  .top-menu-search--loading
+    top: $header-height - 11px  // display directly under ng-input field
+    height: 46px                // ng-option height + 1px border
+    z-index: 1051
+
+  .ng-select#q
+    width: $search-input-width
+    font-size: 0.9rem
+    -webkit-transition: width 0.2s ease-in-out
+    transition: width 0.2s ease-in-out
+    &::-ms-clear
+      margin-right: 5px
+      width: 20px
+
+    .ng-select-container
+      background: transparent
+      border-width: 1px
+      border-style: solid
+      @include varprop(border-color, header-item-font-color)
+      max-height: 32px
+      min-height: unset
+      &:hover
+        @include varprop(border-color, header-item-font-hover-color)
+
+    .ng-arrow-wrapper
+      opacity: 0
+
+    .ng-clear-wrapper
+      @include varprop(color, header-item-font-color)
+      top: 1px
 
-.search-autocomplete--results
-  width: 300px
+    .ng-input
+      top: 0
+      input
+        @include varprop(color, header-item-font-color)
+        height: 30px
+        cursor: text
+
+    .ng-placeholder
+      @include varprop(color, header-item-font-color)
+
+    &.-expanded
+      width: $search-input-width-expanded
+      background: white
+      border-radius: 4px
+      color: $header-search-field-font-color
+
+      .ng-placeholder,
+      .ng-clear-wrapper,
+      input
+        @include varprop(color, header-search-field-font-color)
+
+      .scroll-host
+        @include styled-scroll-bar-vertical
+        max-height: 80vh
+        height: auto
+
+        .ng-option
+          border-bottom: 1px solid #EAEAEA
+          white-space: normal
+          padding: 5px 10px
+          &:last-child
+            border-bottom: none
+          &.ng-option-marked
+            @include varprop(background-color, drop-down-hover-bg-color)
+          &.ng-option-disabled
+            display: none
+
+        .search-autocomplete--wp-id
```

Comment:
The idea of having such long class names is that you do not need to nest them as the nesting is done through the name. If you nest less then it becomes easier to overwrite styles at a later point of time.

Reply:
Please try to avoid nesting unless necessary. I am not a pro yet but have look at BEM to get the general idea:
http://getbem.com/naming/

Reply:
Reading BEM again I probably should have called the class `global-search__wp-id`.

Reply:
Due to the consistency to the rest of our class names, I changed all class names, including yours, following the scheme `global-search--element`.